kucher/view/main_window/register_view_widget/import_export_dialog.py

```diff
@@ -0,0 +1,310 @@
+                    return False
+
+                elif not check_type(reg_check, imported_registers[reg_check.name]):
+                    show_error('Import failed',
+                               f'Cannot import {file_name}',
+                               f'This parameter type is incorrect:\n{reg_check.name}',
+                               parent)
+                    file.close()
+                    _logger.exception(f'Import failed: this parameter type is incorrect {reg_check}')
+                    return False
+
+                if len(imported_registers[reg_check.name]) != len(reg_check.cached_value):
+                    show_error('Import failed',
+                               f'Cannot import {file_name}',
+                               f'This parameter dimension is incorrect:\n{reg_check.name}',
+                               parent)
+                    file.close()
+                    _logger.exception(f'Import failed: this parameter dimension is incorrect {reg_check}')
+                    return False
+
+                elif reg_check.has_min_and_max_values and reg_check.type_id != Register.ValueType.BOOLEAN:
+                    if not (reg_check.min_value <= imported_registers[reg_check.name] <= reg_check.max_value):
+                        show_error('Import failed',
+                                   f'Cannot import {file_name}',
+                                   f'This parameter value is outside permitted range:\n{reg_check.name}',
+                                   parent)
+                        file.close()
+                        _logger.exception('Import failed: this parameter value is outside '
+                                          f'permitted range: {reg_check}')
+                        return False
+
+    except Exception as ex:
+        show_error('Import failed',
+                   f'Cannot import {file_name}',
+                   f'Reason: {str(ex)}',
+                   parent)
+        _logger.exception(f'Could not write registers: {ex!r}')
+        file.close()
+        return False
+    return True
+
+    file.close()
+
+
+def write_registers(parent, file_name, registers, imported_registers):
+    async def executor():
+        unwritten_registers = []
+        changed_registers_number = 0
+        if imported_registers:
+            for reg in imported_registers:
+                for _attempt in range(3):
+                    try:
+                        old_reg = next((r for r in registers if r.name == reg))
+                        new_value = imported_registers[reg]
+                        old_value = await old_reg.read_through()
+                        equal = is_equal(old_reg.type_id, old_value, imported_registers[reg])
+
+                        if not equal:
+                            await old_reg.write_through(new_value)
+                            changed_registers_number += 1
+                        break
```

Comment:
The logic seems unnecessarily complicated. Can we not just write directly instead of reading and comparing first?

Reply:
We can, but we won't have the "xx register edited, yy left unchanged" message

Reply:
Ah, I see. I think this feature is safe to drop at least for now until the basic feature set is implemented and merged.